modules/nextflow/src/main/groovy/nextflow/script/WorkflowDef.groovy

```diff
@@ -254,3 +274,116 @@ class WorkflowParamsResolver {
         return opts
     }
 }
+
+/**
+ * Implements the DSL for publishing workflow outputs
+ *
+ * @author Ben Sherman <bentshermann@gmail.com>
+ */
+@CompileStatic
+class WorkflowPublishDsl {
+
+    private static final List<String> PUBLISH_OPTIONS = List.of(
+        'contentType',
+        'enabled',
+        'failOnError',
+        'mode',
+        'overwrite',
+        'pattern',
+        'saveAs',
+        'storageClass',
+        'tags'
+    )
+
+    private Binding binding
+
+    private Path directory = (Global.session as Session).outputDir
```

Comment:
@pditommaso I assume we want the CLI option / config option to override the output directory defined in the output DSL. But this is tricky because the output directory has already been resolved and sent to the workflow metadata (`workflow.outputDir`). Not sure what the best way would be to handle this correctly.

Reply:
It is unusual to have a runtime setting like this be configurable in the pipeline script. For example the work directory can only be set on the command line or in the config, then in the script it is assumed to be constant

If we want to treat the output directory as a core runtime setting, maybe it would be better to think of the output DSL as being relative to the output directory without needing to specify it explicitly in the DSL.

It will be overridden 99% of the time anyway, and if the user wants to provide a default value for the pipeline, they can set `outputDir` in the config file

Reply:
I don't think it's expected to control the overwrite setting via the CLI, only via config file and script

Reply:
It's pretty common for people to set `params.outdir` from the command line:
```bash
nextflow run nf-core/rnaseq --outdir results ...
```

Reply:
Whereas they would never set it in the script because it makes more sense to put the default in the config file

Reply:
@ewels could use your opinion here

Reply:
it's enough keep using using 

```
output {
  directory params.outdir 
  ...
}
```

Reply:
I thought the point of this was to remove `params.outdir`. If you have it as a CLI option and config option then you don't need a separate param

Reply:
My point is that it's not blocking right now. It could be added in a later iteration

Reply:
Okay then let's not bother with a CLI or config option yet, just the DSL piece and see if users form an opinion on it

I have a feeling that this `directory` method won't be used long-term if there is a CLI + config option, but it doesn't hurt anything if unused